Cython/Compiler/ExprNodes.py

```diff
@@ -6632,16 +6632,68 @@ class PyMethodCallNode(CallNode):
     # arg_tuple   TupleNode     the arguments for the args tuple
     # kwdict      ExprNode or None  keyword dictionary (if present)
     # unpack      bool
+    # function_obj ExprNode or None   copy of self.function.obj (in some situations)    
 
-    subexprs = ['function', 'arg_tuple', 'kwdict']
+    subexprs = ['function', 'arg_tuple', 'kwdict', 'function_obj']
     is_temp = True
     kwdict = None
+    direct_fallback = False
+    function_obj = None
+
+    def __init__(self, pos, **kwds):
+        super().__init__(pos, **kwds)
+        if self.allow_direct_fallback_call():
+            self.direct_fallback = True
+            self.function_obj = ProxyNode(self.function.obj)
+            self.function.obj = CloneNode(self.function_obj)
+
+    def allow_direct_fallback_call(self):
+        if self.kwdict and not isinstance(self.kwdict, DictNode):
+            return False
+        if not self.function.is_attribute:
+            return False
+        return True
+    
+    def generate_direct_fallback_guard(self, code, negate):
```

Comment:
The default for `negate` in all cases outside of the closely fallback related code seems to be True. That suggests that all other code shouldn't pass a flag and the one exception should be the only special place, preferably not using the same method. I'd consider moving the actual C condition into a class level constant and then writing it from two different places. Maybe call it `FALLBACK_CASE` so that you can write `f"#if {FALLBACK_CASE}"`.

Reply:
It looks like it's worth doing this generally (and not just in the limited API) so I'm going to take out these guards and just generate one set of code